server/lib/routes/post-metrics.js

```diff
@@ -19,19 +20,28 @@ module.exports = function () {
     process: function (req, res) {
       // don't wait around to send a response.
       res.json({ success: true });
+
       process.nextTick(function () {
-        var metrics = req.body || {};
-        metrics.agent = req.get('user-agent');
+        try {
+          var metrics = req.body || {};
 
-        if (metrics.isSampledUser) {
-          metricsCollector.write(metrics);
-          // send the metrics body to the StatsD collector for processing
-          statsd.write(metrics);
-        }
+          var contentType = req.get('content-type') || '';
+          if (contentType.indexOf('text/plain') === 0) {
+            metrics = JSON.parse(metrics);
+          }
 
-        ga.write(metrics);
-      });
+          metrics.agent = req.get('user-agent');
 
+          if (metrics.isSampledUser) {
+            metricsCollector.write(metrics);
+            // send the metrics body to the StatsD collector for processing
+            statsd.write(metrics);
+          }
+          ga.write(metrics);
+        } catch (error) {
+          logger.error(error);
```

Comment:
> I don't think the try/catch is needed in the universal sense, just around the try/catch. 

Who wants a try/catch around their try/catch?

Sorry, try/catch around JSON.parse. If an exception occurs, log the error and return.